trace_test.go

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpc
+
+import (
+	"testing"
+)
+
+func TestMethodFamily(t *testing.T) {
+	cases := []struct {
+		Desc               string
+		Method             string
+		WantedMethodFamily string
+	}{
+		{
+			Desc:               "No leading slash",
+			Method:             "pkg.service/method",
+			WantedMethodFamily: "pkg.service",
+		},
+		{
+			Desc:               "Leading slash",
+			Method:             "/pkg.service/method",
+			WantedMethodFamily: "pkg.service",
+		},
+	}
+
+	for _, ut := range cases {
+		got := methodFamily(ut.Method)
+		if got != ut.WantedMethodFamily {
+			t.Fatalf("Desc: %v. Want method family: %v, got: %v", ut.Desc, ut.WantedMethodFamily, got)
```

Comment:
This could be
```
t.Fatalf("methodFamily(%s) = %s, want %s", ut.method, got, ut.wantedMethodFamily)
```
Again, you wouldn't have to specify the testcase description in this error message if you use `t.Run`